### PR TITLE
Fixes #553

### LINKED ISSUE #553: bug: if gcp is selected in the Quickstart Script, and a user has an AWS access key env-var, the script will setup an AWS ManagedZone + secret, rather than a GCP one
Noticed while testing https://github.com/Kuadrant/multicluster-gateway-controller/issues/480

If gcp is selected in the Quickstart Script, and a user has an AWS access key env-var, the script will setup an AWS ManagedZone + secret, rather than a GCP one. Chatted with @R-Lawton about it - reckon its because the initial selection context gets lost between scripts,. The script that sets up the zones + secrets defaults to AWS if a `MGC_AWS_ACCESS_KEY_ID` env var is detected.

### PR DESCRIPTION
**Pull request by jasonmadigan**
Attempting to fix https://github.com/Kuadrant/multicluster-gateway-controller/issues/553

### CODE REVIEW

File: hack/quickstart-setup.sh [hunk 1]
```diff
@@ -23,12 +23,25 @@ if [ -z $MGC_ACCOUNT ]; then
 MGC_ACCOUNT=${MGC_ACCOUNT:="kuadrant"}
 fi
 
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/${MGC_ACCOUNT}/multicluster-gateway-controller/${MGC_BRANCH}/hack/.quickstartEnv)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/${MGC_ACCOUNT}/multicluster-gateway-controller/${MGC_BRANCH}/hack/.kindUtils)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/${MGC_ACCOUNT}/multicluster-gateway-controller/${MGC_BRANCH}/hack/.cleanupUtils)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/${MGC_ACCOUNT}/multicluster-gateway-controller/${MGC_BRANCH}/hack/.deployUtils)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/${MGC_ACCOUNT}/multicluster-gateway-controller/${MGC_BRANCH}/hack/.startUtils)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/${MGC_ACCOUNT}/multicluster-gateway-controller/${MGC_BRANCH}/hack/.setupEnv)"
+if [ -n "$MGC_LOCAL_QUICKSTART_SCRIPTS_MODE" ]; then
```
Comment (jasonmadigan) [inline hack/quickstart-setup.sh:26]: Added this for us devs - makes testing local changes on disk a bit easier.

**Comment by jasonmadigan** [comment]
@R-Lawton WDYT?

File: hack/.deployUtils [hunk 2]
```diff
@@ -327,8 +327,10 @@ configureController() {
     kubectl config use-context kind-${clusterName}
     echo "Initialize local dev setup for the controller on ${clusterName}"
 
-  if [ -n "$MGC_AWS_ACCESS_KEY_ID" ]; then
-    cat <<EOF | kubectl apply -f -
+    if [ "$DNS_PROVIDER" == "aws" ]; then
+        if [ -n "$MGC_AWS_ACCESS_KEY_ID" ]; then
```
Comment (maleck13) [inline hack/.deployUtils:331]: adding validation here might not be bad thing

```
if [ "$MGC_AWS_ACCESS_KEY_ID" == "" ]; then
     echo "MGC_AWS_ACCESS_KEY_ID is not set"
```
etc

Comment (maleck13) [reply hack/.deployUtils:331]: Actually, perhaps a small refactor here overall would be better?

like a switch and some functions?
```
case $DNS_PROVIDER in
   aws)
      echo "setting up aws dns provider"
      awsProvider()
      ;;
   gcp)
      echo "setting up gcp dns provider"
      gcpProvider()
      ;;
   *)
     echo "unknown provider"
     exit
     ;;
esac

```

Comment (mikenairn) [reply hack/.deployUtils:331]: My 2cents, we should not be relying on environment variables anywhere. Its just a bad idea and something i mentioned we should try and avoid previously.

Comment (mikenairn) [reply hack/.deployUtils:331]: > My 2cents, we should not be relying on environment variables anywhere. Its just a bad idea and something i mentioned we should try and avoid previously.

That said, we are unlikely to change how the quick start works at this stage in the game, so what is being proposed makes sense.

**Comment by R-Lawton** [comment]
current changes work locally for me but i agree with the suggested changes for Craig

File: hack/.deployUtils (hunk 2 above)
Comment (jasonmadigan) [reply hack/.deployUtils]: Yeah, I'd agree with that principle too - but, like you allude to, I think we'd have to unwind most of our quickstart to do so. bash isn't great for passing args/params around generally given its nature

Have refactored configureController a bit to use a switch instead

File: hack/.deployUtils [hunk 3]
```diff
@@ -406,9 +395,35 @@ spec:
...
 
+configureController() {
+    clusterName=${1}
+    kubectl config use-context kind-${clusterName}
+    echo "Initialize local dev setup for the controller on ${clusterName}"
+
+    case $DNS_PROVIDER in
+      aws)
+          echo "setting up aws dns provider"
+          if [ "$MGC_AWS_ACCESS_KEY_ID" == "" ]; then
```
Comment (maleck13) [inline hack/.deployUtils:408]: I would put this in the function tbh

**Review by maleck13 — APPROVED** [review]
minor change suggested.  I haven't tried this locally. Don't want to tear down env currently as fixing differnet issue

**Comment by jasonmadigan** [comment]
@maleck13 fixed (agreed). Will do one more quick spin for both providers locally, then merge if all OK?

**Pull request merged**